src/python_framework_v02/troute/nhd_network.py

```diff
@@ -419,3 +426,86 @@ def replace_waterbodies_connections(connections, waterbodies):
             # copy to new network unchanged
             new_conn[n] = connections[n]
     return new_conn
+
+
+def build_subnetworks(connections, rconn, min_size, sources=None):
+    """
+    Construct subnetworks using a truncated breadth-first-search
+    
+    Arguments:
+        connections
+        rconn
+        max_depth
+        sources
+    Returns:
+        subnetwork_master
+    """
+    # if no sources provided, use tailwaters
+    if sources is None:
+        # identify tailwaters
+        sources = headwaters(rconn)
+
+    # create a list of all headwaters in the network
+    all_hws = headwaters(connections)
+
+    subnetwork_master = {}
+    for net in sources:
+
+        # subnetwork creation using a breadth first search restricted by maximum allowable depth
+        # new_sources_list = [net]
+        new_sources = set([net])
+        subnetworks = {}
+        group_order = 0
+        while new_sources:
+
+            # Build dict object containing reachable nodes within max_depth from each source in new_sources
+            rv = {}
+            for h in new_sources:
+
+                reachable = set()
+                Q = deque([(h, 0)])
+                stop_depth = 1000000
+                while Q:
+
+                    x, y = Q.popleft()
```

Comment:
Minor nitpick, more descriptive variable names than `x` and `y` would help make this logic clearer.
the first element of the tuple is the node and the second element represents its depth. I would suggest renaming y -> us_depth and x -> node.

Reply:
```python
node, depth = Q.popleft()
reachable.add(node)
...
```
This also removes the need to set `us_depth` inside the loop.
```python
if len(rx) > 1:
    depth += 1
```

Reply:
Created issue, will address in PR focused on refactor of build_subnetworks function.